magenta-lib/src/main/scala/magenta/deployment_type/Lambda.scala

```diff
@@ -76,19 +74,19 @@ object Lambda extends DeploymentType  {
 
   def defaultActions = List("uploadLambda", "updateLambda")
 
-  def lambdaToProcess(pkg: DeploymentPackage, stage: String, stack: Stack)(reporter: DeployReporter): List[LambdaFunction] = {
+  def lambdaToProcess(pkg: DeploymentPackage, stage: String, stack: Stack, targetRegion: Region)(reporter: DeployReporter): List[LambdaFunction] = {
     val bucketOption = bucketParam.get(pkg)
     if (bucketOption.isEmpty) reporter.warning(s"DEPRECATED: Uploading directly to lambda is deprecated (it is dangerous for CloudFormed lambdas). Specify the bucket parameter and call both uploadLambda and updateLambda to upload via S3.")
     
-    val regions: List[Region] = regionsParam(pkg).flatMap(name => toRegion(name)(reporter))
+    val regions: List[Region] = regionsParam.get(pkg).map(_.filter(regionExists(_)(reporter)).map(Region)).getOrElse(List(targetRegion))
```

Comment:
I was retaining the existing behaviour, but actually changing it makes a lot of sense.